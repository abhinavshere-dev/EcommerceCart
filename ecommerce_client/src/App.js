import axios from "axios";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import "./App.css";
import GenerateCoupan from "./Shopper/GenerateCoupan";
import { Cart } from "./Shopper/cart";
import { Shopper } from "./Shopper/shopper";

axios.defaults.baseURL = process.env.REACT_APP_BASE_URL;
function App() {
  return (
    <BrowserRouter>
      <Routes>  
        <Route path="/" element={<Shopper />} />
        <Route path="/cart" element={<Cart />} />
        <Route path="/generate-coupan" element={<GenerateCoupan />} />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
