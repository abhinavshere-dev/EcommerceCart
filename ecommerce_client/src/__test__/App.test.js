import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('App component matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
