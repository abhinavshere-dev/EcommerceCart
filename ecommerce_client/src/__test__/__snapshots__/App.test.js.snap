// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid p-0"
  >
    <header
      class="flex items-center justify-between p-2 bg-dark text-white bg-black"
    >
      <div
        class="cursor-pointer h3 text-lg"
        data-testid="shopper"
      >
        Shopper
      </div>
      <div
        class="flex gap-4"
      >
        <p
          class="cursor-pointer"
          data-testid="generate-coupan"
        >
          Generate Coupan
        </p>
        <button
          class="bg-gray border rounded-md px-2 py-1 relative"
        >
          Your Cart
          <span
            class="bi bi-cart4"
          />
          <span
            class="absolute top-[-5px] end-[-5px] bg-[red] rounded-[50%] w-[13px] h-[13px] text-[10px] flex justify-center items-center font-extrabold"
          >
            0
          </span>
        </button>
      </div>
    </header>
    <section
      class="mt-3 row"
    >
      <main
        class="col-12 flex flex-wrap overflow-auto justify-around"
        style="height: 100%;"
      />
    </section>
  </div>
</DocumentFragment>
`;
